VideoPlayer Component (#373)
primer/brand: apps/docs/content/components/VideoPlayer.mdx

---
title: VideoPlayer
status: Experimental
source: https://github.com/primer/brand/blob/main/packages/react/src/VideoPlayer/VideoPlayer.tsx
storybook: '/brand/storybook/?path=/story/components-videoplayer--playground'
description: Use the video player component to present videos availible via CDN or local.
---

import {Label} from '@primer/react'

```js
import {VideoPlayer} from '@primer/react-brand'
```

> ## Examples

Reviewer: FYI there seems to be a strange hover / overflow related bug in the docs examples 👇 

https://github.com/primer/brand/assets/13340707/cab69536-23fd-48d5-9c61-3b43e9797f33

Reviewer: Hiding the overflow seems to clip the tooltip. Is there any other workaround? Do we know why that was occurring on hover to begin with? 

![Screenshot 2023-08-11 at 17 13 36](https://github.com/primer/brand/assets/13340707/12528c3e-a55e-4b2d-93c3-f0edf9838aa5)

Author: It seemed to happen because of the tooltip. I think if we want to address this correctly we would need to create a Tooltip component as we have discussed. Then this can be updated.

Reviewer: Creating a dedicated tooltip will require considerable effort, so can we offset this particular tooltip from the left by a few pixels in the meantime?

### Default

```jsx live

> <VideoPlayer title="HLS On Demand Demo" crossOrigin="anonymous">

Reviewer: `crossOrigin` isn't a required attribute, and I don't think we should be opinionated about the value `anonymous` as end-user CORS expectations could differ. 

```suggestion
<VideoPlayer title="HLS On Demand Demo">
```

Author: This was specifically to make the demo work. Ideally, I will have some demo fixtures uploaded to our public folder so we can use those without this attribute.

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)

>   <VideoPlayer.Source src="https://cdn.api.video/vod/vi36GGvd6PoTqViQLxBWwHjJ/mp4/1080/source.mp4" />

Reviewer: Do you know who owns this placeholder video? I think we should try to source our own. Brand Studio may have something that we can use.

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)

  <VideoPlayer.Track

>     src="https://raw.githubusercontent.com/JoshBowdenConcepts/Subtitles-Example/main/sample.vtt"

Reviewer: This is pointing at your personal account right now, could we use a GitHub/Primer-hosted version instead please?

Reviewer: Same also applies for all other instances 🙏

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/a15991caf8ad89311382ef5e27a02f6acfcc13c7)

    default
    kind="subtitles"
    srcLang="en"
    label="English"
  />
</VideoPlayer>
```

## With poster

```jsx live
<VideoPlayer
  poster="https://github.githubassets.com/images/icons/media-player/poster.jpg"
  title="HLS On Demand Demo"

>   crossOrigin="anonymous"

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)

>
  <VideoPlayer.Source src="https://cdn.api.video/vod/vi36GGvd6PoTqViQLxBWwHjJ/mp4/1080/source.mp4" />
  <VideoPlayer.Track
    src="https://raw.githubusercontent.com/JoshBowdenConcepts/Subtitles-Example/main/sample.vtt"
    default
    kind="subtitles"
    srcLang="en"
    label="English"
  />
</VideoPlayer>
```

## Without branding

```jsx live
<VideoPlayer
  branding={false}
  title="HLS On Demand Demo"

>   crossOrigin="anonymous"

Reviewer: ```suggestion
```

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)

>
  <VideoPlayer.Source src="https://cdn.api.video/vod/vi36GGvd6PoTqViQLxBWwHjJ/mp4/1080/source.mp4" />
  <VideoPlayer.Track
    src="https://raw.githubusercontent.com/JoshBowdenConcepts/Subtitles-Example/main/sample.vtt"
    default
    kind="subtitles"
    srcLang="en"
    label="English"
  />
</VideoPlayer>
```

## Component props

<h3>
  VideoPlayer <Label>Required</Label>
</h3>

`VideoPlayer` provides a React alternative to the native HTML `<video />`.

The component API supports all standard HTML attribute props, while providing some additional behavior as described below.

> 

Reviewer: For consistency with other docs pages, could you please drop these details to underneath its relevant prop table?

Author: Sure! It was before the prop table for Checkbox which I grabbed this language from should I update Checkbox to reflect this change as well?

Reviewer: Checkbox will eventually need to be updated, but can be done separately. They should go underneath the main table. Examples: [SectionIntro](https://primer-bb0111d97b-26139705.drafts.github.io/components/SectionIntro/react#component-props), [Card](https://primer-bb0111d97b-26139705.drafts.github.io/components/Card/react#cardheading), [ComparisonTable](https://primer-bb0111d97b-26139705.drafts.github.io/components/ComparisonTable#comparisontablefootnote), etc

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/0ec9c6c62a4ab24513059c86e087fb4cd7728a33)

| Name        | Type                                   | Default | Required | Description                                                          |
| :---------- | :------------------------------------- | :-----: | :------: | :------------------------------------------------------------------- |

> | `title`     | `boolean`                              |         |  `true`  | Sets the title of the video                                          |

Reviewer: Should this be a `string` type instead of boolean?

```suggestion
| `title`     | `string`                              |         |  `true`  | Sets the title of the video                                          |
```

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)

> | `children`  | `React.ReactNode`, `React.ReactNode[]` |         |  `true`  | Takes `<source>` and `<track>` elements to present video information |

Reviewer: nit: I don't think <track> is required, so may need to call that out in the description.

Author: I make this property required because it will be required for accessibility. Are you saying we should update it to not be required or is there another way you are looking for me to represent this?

Author: This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/0ec9c6c62a4ab24513059c86e087fb4cd7728a33)

---
title: Video player
status: Experimental
source: https://github.com/primer/brand/blob/main/packages/react/src/VideoPlayer/VideoPlayer.tsx
storybook: '/brand/storybook/?path=/story/components-videoplayer--playground'
description: Use the video player component to enable playback of self-hosted videos
---

import {Label} from '@primer/react'

```js
import {VideoPlayer} from '@primer/react-brand'
```

## Examples

### Default

```jsx live
<VideoPlayer title="GitHub media player">
  <VideoPlayer.Source src="/example.mp4" />
  <VideoPlayer.Track src="/example.vtt" />
</VideoPlayer>
```

## With poster

```jsx live

> <VideoPlayer poster="/example-poster.png" title="GitHub Video Demo">

Reviewer: Update to title for consistency with previous example

```suggestion
<VideoPlayer poster="/example-poster.png" title="GitHub media player">
```

Author: Fixed in the following [SHA](https://github.com/primer/brand/pull/373/commits/ee6ae80b15875236b720a51258e132c66f39e856)

  <VideoPlayer.Source src="/example.mp4" />
  <VideoPlayer.Track src="/example.vtt" />
</VideoPlayer>
```

## Without branding

```jsx live

> <VideoPlayer branding={false} title="GitHub Video Demo">

Reviewer: Update to title for consistency with previous example

```suggestion
<VideoPlayer branding={false} title="GitHub media player">
```

Author: Fixed in the following [SHA](https://github.com/primer/brand/pull/373/commits/ee6ae80b15875236b720a51258e132c66f39e856)

| `src`  | `string` |         |  `true`  | Sets the URL of the video content     |
| `type` | `string` |         | `false`  | Sets media type for the video content |

<h3>VideoPlayer.Track</h3>

`VideoPlayer.Track` provides a React alternative to the native HTML `<track />`.

The component API supports all standard HTML attribute props, while providing some additional behavior as described below.

| Name      | Type                                                            | Default | Required | Description                                                                                                                              |
| :-------- | :-------------------------------------------------------------- | :-----: | :------: | :--------------------------------------------------------------------------------------------------------------------------------------- |
| `src`     | `string`                                                        |         |  `true`  | Sets the source of the VTT file containing captions                                                                                      |

> | `kind`    | `subtitles`, `captions`, `descriptions`, `chapters`, `metadata` |         | `false`  | Sets how the text track is meant to be used                                                                                              |

Reviewer: These types should have quotes around them to indicate they are string types.

```suggestion
| `'kind'`    | `'subtitles'`, `'captions'`, `'descriptions'`, `'chapters'`, `'metadata'` |         | `undefined`  | Sets how the text track is meant to be used                                                                                              |
```

Author: Fixed in the following [SHA](https://github.com/primer/brand/pull/373/commits/ee6ae80b15875236b720a51258e132c66f39e856)

---

> title: VideoPlayer

Reviewer: ```suggestion
title: Video player
```

Reviewer: ICYMI all docs page headers now use sentence case

status: Experimental
source: https://github.com/primer/brand/blob/main/packages/react/src/VideoPlayer/VideoPlayer.tsx
storybook: '/brand/storybook/?path=/story/components-videoplayer--playground'

> description: Use the video player component to present videos availible via CDN or local.

Reviewer: ending at local feels a bit abrupt, how about this suggestion for conciseness...

```suggestion
description: Use the video player component to present self-hosted videos available via CDN or local.
```

---

import {Label} from '@primer/react'

```js
import {VideoPlayer} from '@primer/react-brand'
```

## Examples

### Default

```jsx live

> <VideoPlayer title="HLS On Demand Demo">

Reviewer: Thoughts on changing this title to be more generic?

```suggestion
title="GitHub media player"
```

The component API supports all standard HTML attribute props, while providing some additional behavior as described below.

| Name   | Type     | Default | Required | Description                           |
| :----- | :------- | :-----: | :------: | :------------------------------------ |
| `src`  | `string` |         |  `true`  | Sets the URL of the video content     |
| `type` | `string` |         | `false`  | Sets media type for the video content |

<h3>VideoPlayer.Track</h3>

`VideoPlayer.Track` provides a React alternative to the native HTML `<track />`.

The component API supports all standard HTML attribute props, while providing some additional behavior as described below.

| Name      | Type                                                            | Default | Required | Description                                                                                                                              |
| :-------- | :-------------------------------------------------------------- | :-----: | :------: | :--------------------------------------------------------------------------------------------------------------------------------------- |

> | `src`     | `boolean`                                                       |         |  `true`  | Sets the source of the VTT file containing captions                                                                                      |

Reviewer 2: ```suggestion
| `src`     | `string`                                                       |         |  `true`  | Sets the source of the VTT file containing captions                                                                                      |
```
